src/main/http.ts

```diff
@@ -12,17 +11,12 @@ import conditional from 'koa-conditional-get';
 import etag from 'koa-etag';
 import cors from '@koa/cors';
 import * as middleware from './middleware';
-
-const PORT = numberConfig('PORT', 8080);
-const HTTP_TIMEOUT = numberConfig('HTTP_TIMEOUT', 300000);
-const HTTP_SHUTDOWN_DELAY = numberConfig('HTTP_SHUTDOWN_DELAY', 10000);
+import * as env from './env';
```

Comment:
May I suggest that we extract all the `readNumber`, `readString` etc in a separate file to make these configs more in line with how app is configured?

We can also provide a `BaseEnv` with those, and apps would inherit their envs from it.

Reply:
Thanks 👍 would you mind elaborate it in more details please,
What I understood is `BaseEnv` should be injectable and we should bind it to an app so that it can be used in `node-framework`'s services, for example here.

```ts
@injectable()
export class BaseEnv {
    PORT = readNumber('PORT', 8080);
    HTTP_TIMEOUT = readNumber('HTTP_TIMEOUT', 300000);
    HTTP_SHUTDOWN_DELAY = readNumber('HTTP_SHUTDOWN_DELAY', 10000);

    constructor() {
        // Including this will throw an error if at least one of the envs
        // is undefined at application start
        assertEnv();
    }
}
```
so it will be similar to how we used `Configuration` previously?

Reply:
Yep, so [docs here](https://github.com/ubio/node-framework/blob/master/docs/env.md) suggest that each application creates its own env and consume it as they see fit.

What we can do is create a tiny class (essentially what you've specified above, except that `assertEnv` and constructor aren't really necessary since all properties come with defaults), then bind it to container inside `App` (in framework) and then use it inside framework's components to resolve those settings. It can be called `FrameworkEnv` to avoid confusion. If necessary, we can have a 🤙